### PR TITLE
fix(docs) Do not polyfill node bindings by default

### LINKED ISSUE #2706: Document webpack change: Do not polyfill node bindings by default
*A pull request by @ooflorent was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/8460

---

**What kind of change does this PR introduce?**

refactoring

**Did you add tests for your changes?**

yes

**Does this PR introduce a breaking change?**

yes

**What needs to be documented once your changes are merged?**

1. `node` has fewer options:
    - `global`
    - `__filename`
    - `__dirname`

2. `Buffer`, `setImmediate`, `process` and `console` are no longer polyfilled
3. `global` is still supported but is disabled by default (EDIT: reverted)
4. `require("vm")` or other node builtin packages are no longer polyfilled.  
Use `resolve.fallback` instead.

### PR DESCRIPTION
**Pull request by byzyk**
Fixes #2706

### CODE REVIEW

**Comment by byzyk** [comment]
Hi @sokra, can you please give a small clarification on point 4 from the issue:

> require("vm") or other node builtin packages are no longer polyfilled.
> Use resolve.alias instead.

Does it mean we need to remove [Other core libs](https://webpack.js.org/configuration/node/#other-node-core-libraries) section at all and add a note about `resolve.alias`?

File: src/content/configuration/node.md [hunk 1]
```diff
@@ -100,20 +80,6 @@ Options:
...
-
-
-## `node.setImmediate`
-
-`boolean | "mock" | "empty"`
-
-Default: `true`
-
-
 ## Other node core libraries
```
Comment (sokra) [inline src/content/configuration/node.md]: This should also be removed

File: src/content/configuration/node.md [hunk 2]
```diff
@@ -33,13 +33,9 @@ These are the defaults:
...
-    console: false,
-    global: true,
-    process: true,
+    global: false,
     __filename: 'mock',
     __dirname: 'mock',
-    Buffer: true,
-    setImmediate: true
 
     // See "Other node core libraries" for additional options.
```
Comment (sokra) [inline src/content/configuration/node.md]: Remove this line

**Comment by EugeneHlushko** [comment]
Thanks!

**Pull request merged**